src/observe-tag.js

```diff
@@ -56,19 +82,35 @@ module.exports = tagFunction => {
 
 			// if an element had focus, reapply it
 			if (removedElementWithFocusData.index >= 0) {
-				const children = tagResult.querySelectorAll('*')
-
-				const elementToGiveFocus = [tagResult, ...children][removedElementWithFocusData.index]
-				elementToGiveFocus.focus()
-
-				// also try to set the selection, if there is a selection for this element
-				const hasSelectionStart = removedElementWithFocusData.selectionStart !== null && removedElementWithFocusData.selectionStart !== undefined
-				if (hasSelectionStart) {
-					elementToGiveFocus.setSelectionRange(
-						removedElementWithFocusData.selectionStart,
-						removedElementWithFocusData.selectionEnd,
-						removedElementWithFocusData.selectionDirection
-					)
+				const allElements = parentAndChildrenElements(tagResult)
+
+				// we'll look through the elements (in order of nodes closest to original index) and find a tag that matches.
+				// this means if it didn't move, we'll get it right away,
+				// if it did, we'll look at the elements closest to the original position
+				const nodeMatchesTagName = node => node.tagName === removedElementWithFocusData.tagName
+				const elementToGiveFocus = allElements
+					.map(toIndicies)
+					.sort(byDistanceFromIndex(removedElementWithFocusData.index))
+					.map(toNodes(allElements))
+					.find(nodeMatchesTagName)
+
+				// if the element / child exists, focus it
+				if (elementToGiveFocus !== undefined) {
```

Comment:
This is the actual fix. We look through all elements now, and verify that the element even exists. Previously, even if the element was removed, we'd still try to focus on something, which caused errors to be thrown.